openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/enricher/RouteURLEnricher.java

```diff
@@ -76,11 +76,6 @@ private URL lookup(RouteURL routeURL) {
             throw new NullPointerException("CubeOpenShiftConfiguration is null.");
         }
 
-        final String routerAddress = config.getRouterHost();
-        if (routerAddress == null || routerAddress.length() == 0) {
-            throw new IllegalArgumentException("Must specify routerHost!");
-        }
-
         final OpenShiftClient client = clientInstance.get();
         final Route route = client.getClient().routes().inNamespace(config.getNamespace()).withName(routeName).get();
```

Comment:
If routeName is null, then I guess that `OpenShiftClient` just detects a `null` and do nothing isn't it?

Reply:
If `routeName` is `null`, then a NPE is thrown a couple of lines above.

Reply:
So we are getting a `NPE` right? So this is not fixing the error, or am I missing something?

Reply:
So, there are two things: `routeName`, which is the value of the `@RouteURL` annotation, and that is of course mandatory. And then there's this `routerAddress`, which is like totally useless in this case, and it is the check for `routerAddress` that I'm removing here.

Reply:
ah right sorry I miss the names, true, sorry for my confusion. Maybe it would be better a log line? Or this would not add any interesting information for the user. I am just thinking to not missing something important that the developer might be interested with like this.

Reply:
So, frankly, I really don't understand the purpose of configuring the address of the router, as routes are AFAIK always resolvable (that's the purpose of routes, right?). But apparently, there are cases that I'm not aware of, so I need to lean on your advice.

Reply:
@Ladicek the purpose of RouteURL is resolve the routes outside OPenshif for the tests which are annotated with @RunAsClient, it read all routes from Openshift and add the the JVM custom DNS naming service.
If you remove this check when use the @RoutURL it will thrown a NPE later when the tests are running in the time arquillian will try to add the DNS records to the JVM.
If you are not running tests with RunAsClient so there is no reason to you to use this annotation.

Reply:
@Ladicek routes are not always resolvable. It depends on how the cluster administrator set them, or how users create them. Like we keep saying, addresses must be resolvable by somebody. Be it yourself, or your provider dns server, or anybody else.

So, like I said in the past, I'm +1 for this patch.

Reply:
But what @spolti suggests is that this patch might produce a NPE isn't it?

Reply:
Clear now :) Then we are ready to merge isn't it?

Reply:
@spolti Sorry, I don't follow. I'm pretty sure that the change in this PR is actually OK, though, because this branch https://github.com/Ladicek/wfswarm-rest-http/tree/arquillian-cube builds and runs just fine for me. (`mvn clean verify -Popenshift,openshift-it -Dnamespace.use.current=true` is the command to use, in case anyone's interested.) I'm not using `@RunAsClient` there, but I tried it and it doesn't make a difference.

Reply:
@lordofthejars yep

Reply:
@Ladicek yes he tell that he was wrong, that it is currently working as expected

Reply:
Thanks all!